packages/renderer/server/render-video.ts

```diff
@@ -314,62 +333,44 @@ async function cleanup(
   await Promise.all([...folderCleanupPromises, ...fileCleanupPromises]);
 }
 
-export const renderPartialVideo = async (
-  projectFile: string,
-  workerId: number,
-  numWorkers: number,
-  params?: Record<string, unknown>,
-  progressCallback?: (worker: number, progress: number) => void,
-  settings: RenderVideoSettings = {},
-) => {
-  // Get settings
-  const outputFileName = settings.name ?? 'project';
-  const outputFolderName = settings.outDir ?? './output';
-  const hiddenFolderId = uuidv4();
-  const numOfWorkers = numWorkers;
-
-  await initializeBrowserAndStartRendering(
-    projectFile,
-    outputFileName,
-    outputFolderName,
-    workerId,
-    numOfWorkers,
-    settings,
-    hiddenFolderId,
-    params,
-    progressCallback,
-  );
+function getPropDefaults(settings: RenderSettings) {
+  if (settings.outFile && !settings.outFile.endsWith('.mp4')) {
+    throw new Error('outFile must end with ".mp4"');
+  }
 
-  const videoFilePath = `${os.tmpdir()}/revideo-${outputFileName}-${workerId}-${hiddenFolderId}/visuals.mp4`;
-  const audioFilePath = `${os.tmpdir()}/revideo-${outputFileName}-${workerId}-${hiddenFolderId}/audio.wav`;
+  const outFile = settings.outFile?.slice(0, -4) ?? 'video';
 
-  if (!(await doesFileExist(audioFilePath))) {
-    const videoDuration = await getVideoDuration(videoFilePath);
-    await createSilentAudioFile(audioFilePath, videoDuration);
-  }
+  return {
+    outputFileName: outFile,
+    outputFolderName: settings.outDir ?? './output',
+    numOfWorkers: settings.workers ?? 1,
+    hiddenFolderId: uuidv4(),
+  };
+}
 
-  return {audioFile: audioFilePath, videoFile: videoFilePath};
-};
+interface RenderVideoProps {
+  projectFile: string;
+  variables?: Record<string, unknown>;
+  progressCallback?: (worker: number, progress: number) => void;
+  settings?: RenderSettings;
+}
 
 /**
  * Renders a video to a file.
- * @param configFile - Path to the vite config file (vite.config.ts).
- * @param params - Parameters to pass to your project (see https://docs.re.video/parameterized-video)
+ * @param projectFile - Path to the project.ts file.
+ * @param variables - Variables to pass to your project (see https://docs.re.video/parameterized-video)
  * @param progressCallback - Callback function to track rendering progress. Progress is a number between 0 and 1.
  * @param settings - Settings for the rendering process.
  * @returns - Path to the rendered video file.
  */
-export const renderVideo = async (
-  projectFile: string,
-  params?: Record<string, unknown>,
-  progressCallback?: (worker: number, progress: number) => void,
-  settings: RenderVideoSettings = {},
-) => {
-  // Get settings
-  const outputFileName = settings.name ?? 'project';
-  const outputFolderName = settings.outDir ?? './output';
-  const hiddenFolderId = uuidv4();
-  const numOfWorkers = settings.workers ?? 1;
+export const renderVideo = async ({
+  projectFile,
+  variables,
+  progressCallback,
+  settings = {},
```

Comment:
do you think progressCallback should be part of settings or does it not really matter?